### ISSUE TITLE
Unit tests should fail on unexpected "Noncompliant" comments

### ISSUE THREAD

**Issue by alban-auzeill** [SonarSource/sonar-dotnet#2672]
Unit tests are based on this kind of comments:
```cs
  foo(); // Noncompliant {{This method is dangerous}}
```
But humans can write them with errors, and in this case, the test should fail:
```cs
  foo(); // Noncompliant {This method is dangerous}
  foo(); // Noncompliant {"This method is dangerous"}
```
One problem to enforce Noncompliant text format is the fee comments at the end. In this case one solution can be to enforce a specific 'free comment separator' like ' - ':
```cs
  foo(); // Noncompliant - false-positive, limitation of our analyzer
```
